app/questionnaire/path_finder.py

```diff
@@ -286,14 +258,9 @@ def _remove_current_blocks_answers_for_backwards_routing(
                 block_id
             )
             if "when" in rules:
-                if isinstance(rules["when"], dict):
-                    self._remove_current_blocks_answers_for_new_backwards_routing(
-                        rules["when"], answer_ids_for_current_block
-                    )
-                else:
-                    for rule in rules["when"]:
-                        if "id" in rule and rule["id"] in answer_ids_for_current_block:
-                            self.answer_store.remove_answer(rule["id"])
+                self._remove_current_blocks_answers_for_new_backwards_routing(
```

Comment:
Not "new" anymore since we remove old?

Reply:
Renamed it to `_remove_block_anwers_for_backward_routing_according_to_when_rule`
Kept the method as `"when" in rule` might still be False, e.g if the routing block doesn't have a when rule